homeassistant/components/pi_hole/config_flow.py

```diff
@@ -144,3 +141,16 @@ async def _async_try_connect(self, host, location, tls, verify_tls, api_token):
             api_token=api_token,
         )
         await pi_hole.get_data()
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle options."""
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id="init", data_schema=vol.Schema({vol.Optional(CONF_API_KEY): str}),
```

Comment:
Please remove the api key option from the main flow form since we have the option here now.

Reply:
Only via import of config yaml. We don't want optional items in the main flow.

Reply:
I feel it does not make much sense that users cannot set up the API key in the main flow. What about adding a switch indicating if users need switch or read-only and setting up API key in a separated step? This will solve #37259 as well.

Reply:
The api key should either be in the main flow or in the options flow, not in both.

Reply:
Ok. Then I think it makes more sense to be in the main flow. I'll remove the options flow.